rclcpp/include/rclcpp/intra_process_manager_impl.hpp

```diff
@@ -253,8 +256,20 @@ class IntraProcessManagerImpl : public IntraProcessManagerImplBase
   using SubscriptionMap = std::unordered_map<uint64_t, subscription::SubscriptionBase::WeakPtr,
       std::hash<uint64_t>, std::equal_to<uint64_t>,
       RebindAlloc<std::pair<const uint64_t, subscription::SubscriptionBase::WeakPtr>>>;
-  using IDTopicMap = std::map<std::string, AllocSet,
-      std::less<std::string>, RebindAlloc<std::pair<const std::string, AllocSet>>>;
+
+  struct strcmp_wrapper : public std::binary_function<const char *, const char *, bool>
```

Comment:
I don't feel it needs it's own file, it's a pretty trivial class, I would have made it a lambda if I could have. I'd prefer not to separate it, but you can push here with that change if you like.